st2common/st2common/models/db/__init__.py

```diff
@@ -340,6 +340,8 @@ def query(self, *args, **filters):
         limit = filters.pop('limit', None)
         order_by = filters.pop('order_by', None)
         exclude_fields = filters.pop('exclude_fields', None)
+        only_fields = filters.pop('only_fields', None)
```

Comment:
Because of the way our existing `query()` method works we can't simply chain the queryset filter methods - e.g. `query().only('foo').no_dereference()`, but we need to perform that chaining inside the query method.

That's important, because we already evaluate the queryset where we slice the result set a couple of lines below.

That's just more of a heads up / warning, because if you chained the methods like the examples I included above it wouldn't work correctly (it would have no effect) since result is already evaluated inside `query()` method. That "no effect" also wouldn't be visible unless you had a specific test case to catch that scenario.

^ @lakshmi-kannan @m4dcoder @bigmstone @enykeev  Just a heads up. This is not totally clear from the `query()` API (which mostly predates me), I only noticed it because I was using `--profile` flag to make sure `only()` works correctly.